### PR TITLE
Fix copy assignment and the comparison operator of `rmm_host_allocator`

### PR DESCRIPTION
**Pull request by vuule**
## Description
Copy assignment of `rmm_host_allocator`, used in `hostdevice_vector`, is missing the `stream` member assignment, leading to deallocation in the default stream in the assigned-to allocator.

This PR fixes this error by switching to the auto-generated special functions.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by shrshi — APPROVED** [review]
:fire:

File: cpp/include/cudf/detail/utilities/rmm_host_vector.hpp [hunk 1]
```diff
@@ -109,30 +109,6 @@ class rmm_host_allocator {
...
-   * @brief Move constructor
-   */
-  rmm_host_allocator(rmm_host_allocator&& other) = default;
-
-  /**
-   * @brief Assignment operator
-   */
-  rmm_host_allocator& operator=(rmm_host_allocator const& other)
-  {
-    mr = other.mr;
```
Comment (PointKernel) [inline cpp/include/cudf/detail/utilities/rmm_host_vector.hpp:127]: I notice that the equality operator `==` also only checks mr but not stream. Seems it's intentional but not sure why. Worth double checking with @nvdbaranec 

Could we set up a stream mode test for `cpp/tests/utilities_tests/io_utilities_tests.cpp`? In this way, we can avoid/uncover similar issues before delivery.

Comment (vuule) [reply cpp/include/cudf/detail/utilities/rmm_host_vector.hpp:127]: added stream comparison to the operator==

**Review by vyasr — APPROVED** [review]

**Pull request merged**